linux.yml

```diff
@@ -35,7 +35,7 @@
       :environment:
         - CFLAGS=-O3 -fPIC -D_FORTIFY_SOURCE=2 -DWOLFSSL_MIN_RSA_BITS=2048 -DWOLFSSL_MIN_ECC_BITS=256 -DUSE_CERT_BUFFERS_4096 -DUSE_CERT_BUFFERS_256 -DWOLFSSL_NO_SPHINCS -Wno-error=stringop-overflow
       :build:
-        - git apply ../../wolfssl/*.patch
+        - git apply ../../wolfssl/*.patch || true
```

Comment:
I'm not sure if we want to always `II true` and mask any errors, on top of removing the `# For some reason this attempts to run twice as a result the git apply will fail the second time` comment.

In my opinion:
1. Removing the commment makes future readers confused about why `|| true` was done. It's not obvious just from reading the code.
2. Blindly adding `|| true` risks hiding failures with applying patches, which could cause the output binary to silently not include the patch that we were expecting.

Reply:
`git apply ...` exits with 1 if there's no files to patch, so adding `|| true` is the simplest way to workaround it. The previous comment "for some reason..." is no long accurate now, thus it was removed als

If you're worried that `|| true` will mask errors in the future, then I think we should just remove all the `git apply` commands. I'll fix it.

Reply:
Thanks for clarifying Tom!
I'm wondered if I was too paranoid so I dug a bit deeper and looked at `https://stackoverflow.com/questions/4917871/does-git-return-specific-return-error-codes` .  I guess many different errors could cause an exit with 1, so probably best to just not ignore it and hope it was "no files to patch"

Reply:
Fixed.

Reply:
In the future could consider:
```bash
shopt -o nullglob; set -e; for p in ../../wolfssl/*.patch ; do git apply "$p" ; done
```

`nullglob` causes `../../wolfssl/*.patch` to expand to nothing if nothing matches (default unhelpfully gives you `../../wolfssl/*.patch` back in that case...), so this is a nop if there are no patches.

In the past I've had problems on rebuild due to the patches already being applied and trying to apply again, so if we wanted to get proper fancy:

```bash
shopt -o nullglob
set -e

for p in ../../wolfssl/*.patch ; do
    if [ ! -e "$p.applied" ] ; then 
        git apply "$p"
        touch "$p.applied"
done
```

(and an `rm ....wolfssl/*.patch.applied` in a suitable clean target + .a `gitiognore` entry)

Or we could go whole hog and depend on [`quilt(1)`](https://savannah.nongnu.org/projects/quilt)...

Reply:
`Quilt`: what a clever name! A lot of patches eh. :grin: